### PR TITLE
RAD-243 Avoid NPEs in Hibernate DAOs

### PR DESCRIPTION
**Pull request by bgeVam**
## Description
* Change HibernateDAOs to return empty lists instead of null

## Related Issue
see https://issues.openmrs.org/browse/RAD-243

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/radiology/order/HibernateRadiologyOrderDAO.java [hunk 1]
```diff
@@ -55,13 +55,12 @@ public RadiologyOrder getRadiologyOrderByOrderId(Integer orderId) {
...
 	@Override
 	public List<RadiologyOrder> getRadiologyOrdersByPatient(Patient patient) {
-		List<RadiologyOrder> result = new ArrayList<RadiologyOrder>();
 		
 		Criteria radiologyOrderCriteria = createRadiologyOrderCriteria();
 		addRestrictionOnPatient(radiologyOrderCriteria, patient);
 		
-		result = (List<RadiologyOrder>) radiologyOrderCriteria.list();
-		return result;
+		List<RadiologyOrder> result = (List<RadiologyOrder>) radiologyOrderCriteria.list();
```
Comment (teleivo) [inline api/src/main/java/org/openmrs/module/radiology/order/HibernateRadiologyOrderDAO.java]: use final

File: api/src/main/java/org/openmrs/module/radiology/order/HibernateRadiologyOrderDAO.java [hunk 2]
```diff
@@ -92,13 +91,12 @@ private void addRestrictionOnPatient(Criteria criteria, Patient patient) {
...
 	@Override
 	public List<RadiologyOrder> getRadiologyOrdersByPatients(List<Patient> patients) {
-		List<RadiologyOrder> result = new ArrayList<RadiologyOrder>();
 		
 		final Criteria radiologyOrderCriteria = createRadiologyOrderCriteria();
 		addRestrictionOnPatients(radiologyOrderCriteria, patients);
 		
-		result = (List<RadiologyOrder>) radiologyOrderCriteria.list();
-		return result;
+		List<RadiologyOrder> result = (List<RadiologyOrder>) radiologyOrderCriteria.list();
```
Comment (teleivo) [inline api/src/main/java/org/openmrs/module/radiology/order/HibernateRadiologyOrderDAO.java]: use final

File: api/src/main/java/org/openmrs/module/radiology/report/HibernateRadiologyReportDAO.java [hunk 3]
```diff
@@ -91,11 +92,13 @@ public boolean hasRadiologyOrderClaimedRadiologyReport(RadiologyOrder radiologyO
 	@Override
 	public List<RadiologyReport> getRadiologyReportsByRadiologyOrderAndRadiologyReportStatus(RadiologyOrder radiologyOrder,
 			RadiologyReportStatus radiologyReportStatus) {
-		return sessionFactory.getCurrentSession()
+		
+		List<RadiologyReport> result = sessionFactory.getCurrentSession()
```
Comment (teleivo) [inline api/src/main/java/org/openmrs/module/radiology/report/HibernateRadiologyReportDAO.java]: use final

**Comment by bgeVam** [comment]
PR Updated

**Comment by bgeVam** [comment]
PR Rebased

**Pull request merged**